### ISSUE TITLE
Train a seed color prediction model

### ISSUE THREAD

**Issue by karnehen** [karnehen/ColorChecker#8]

**Comment by karnehen** [comment]
API for machine learning models in OpenCV is here - https://docs.opencv.org/java/3.1.0/org/opencv/ml/package-tree.html
here is the example of loading a pretrained model - http://answers.opencv.org/question/92386/svm-load-still-missing-in-java/?answer=92439#post-id-92439

Predicting color looks like a simple problem, so the main issue here is prediction performance. So, neural networks and trees ensembles are most likely out of question

**Comment by karnehen** [comment]
Tried some simple classification algorithms (seedwise) - average accuracy is about 80%. Red seed are better predicted with KNN, other - with logistic regression. It makes sense to try blending them.

Also I'll try to predict pixelwise and average for the seeds

**Comment by karnehen** [comment]
Decided that it should not predict a single color label, but several characterizing features. So I'll do this:
  * try different features (mean, percentiles, std) and see which of them help to predict color better
  * calculate them instead of prediction
  * another feature idea is to do K-means clustering of seed pixels and draw the mean color in each cluster
  * for percentiles I should look on color distributions for single seeds

**Comment by karnehen** [comment]
Merged: https://github.com/karnehen/ColorChecker/pull/20

I've made classification in Python. Had some problems with using the pretrained model in Java, but we decided to do unsupervised learning anyway. The classification sample is thus just to show how separable the seeds are and how much does calibration affect it.
